wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Create/WireTextField.swift

```diff
@@ -0,0 +1,146 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import UIKit
+import WireCommonComponents
+import WireDesign
+
+protocol WireTextFieldDelegate: AnyObject {
+
+    func textField(_ textField: WireTextField, valueChanged value: WireTextField.Value)
+
+    func textFieldReturnPressed(_ textField: WireTextField)
+
+    func textFieldDidEndEditing(_ textField: WireTextField)
+
+    func textFieldDidBeginEditing(_ textField: WireTextField)
+
+}
+
+class WireTextField: UITextField {
```

Comment:
Nice, that is the kind of shared component that could be moved to WireUI package, no ?

Reply:
Since it's being used at exactly one place (at least for now) I wouldn't move it.